sdk/python/lib/pulumi/dynamic/__main__.py

```diff
@@ -83,13 +95,18 @@ def Invoke(self, request, context):
         raise NotImplementedError(f"unknown function {request.token}")
 
     def Diff(self, request, context):
+        _ensure_event_loop()
```

Comment:
I think we can remove this again. This was needed to make the full `pulumi.Config` work here, however this is problematic for the case where the provider is defined in `__main__`, as it makes the dill pickle library attempt to pickle protobufs, and fail.